lm_eval/api/task.py

```diff
@@ -519,15 +520,46 @@ def apply_filters(self):
             return self._instances
 
     def dump_config(self) -> dict:
-        """Returns a dictionary representing the task's config.
-
-        :returns: str
-            The fewshot context.
-        """
+        """Returns the config as a dictionary."""
         # TODO: this should only return the overrides applied to a non-YAML task's configuration.
         # (num_fewshot)
         return self.config.to_dict()
 
+    def set_config(self, key: str, value: Any, update: bool = False) -> None:
+        """Set or update the configuration for a given key."""
+        if key is None:
+            raise ValueError("Key must be provided.")
+
+        if update:
+            current_value = getattr(self._config, key, {})
+            if not isinstance(current_value, dict):
+                raise TypeError(
+                    f"Expected a dict for key '{key}', got {type(current_value).__name__} instead."
+                )
+            current_value.update(value)
+        else:
+            setattr(self._config, key, value)
+
+    def override_metric(self, metric_name: str) -> None:
```

Comment:
When this is called on a `Task` object not inheriting from `ConfigurableTask`, what happens? I'm not sure these will have the `_metric_fn_list` attribute, etc.

Reply:
what task can i try it on?

Reply:
Not completely sure if it will be enough, but Scrolls seems based on Task: https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/tasks/scrolls/task.py

Reply:
Yeah tried that but can't seem to run just one sub-task rather than downloading the whole benchmark.

Reply:
The other option is to undo https://github.com/EleutherAI/lm-evaluation-harness/pull/1384 and try on Squadv2

Reply:
@haileyschoelkopf Added a condition. Works on Squadv2 (`Task`)!